### ISSUE TITLE
Translations: How to?

### ISSUE THREAD

**Issue by freibadschwimmer** [Gremmel/kleiner-brauhelfer#10]
Hi!

First of all, thanks a lot for this great software, helped me with numerous brews already!

Unfortunately I don't know nothing about programming, but still would like to help/contribute, if times allows me. Thus I wanted to ask, what is the way to go if one wants to add a translation to the software?

Is it enough to add the translations to a source/languages/kb_LANGUAGE.ts file?
I see that each translation also has a *.qm file, with which I wouldn't know how to deal with.

In short: if I'd know how, I might be able to contribute.

Cheers!

**Comment by BourgeoisLab** [comment]
Follow this tutorial: 
https://wiki.qt.io/How_to_create_a_multi_language_application#Add_translations_to_the_project

In what language do you want to translate?
If you need help to set it up, just tell me.

**Comment by freibadschwimmer** [comment]
I would like to translate "den kleinen brauhelfer" into English.

So thanks for the hint! Apparently it works a bit differently than I thought :-)
Anyhow, I have installed Qt now and will start getting acquainted with it and how to translate.

Thanks for the offer to help setting it up. I might come back to that.

Cheers!

**Comment by freibadschwimmer** [comment]
Thanks again @BourgeoisLab for your reply.

I think I managed to set up everything, though since I am totally new to Qt (and using github) it would be grant to get some feedback whether or not I'm on the right track.

What I did so far was cloning a copy of the master branch (to my desktop), and importing it as an "existing program" to Qt (Qt Creator 4.2.0 / Qt 5.7.1).
Then, following the link above, I added "languages/kb_en.ts" to the brauhelfer.pro and ran the lupdate command.
This gave me the kb_en.ts file to work with in Qt linguist. Though it also changed a couple of other language files, hope it should be like that.

Before I jump into the translation work, which might take a while anyways, I just wanted to check if I'm on the right track here.

So I have a few questions regarding this:

- Is it OK if I'm working on a clone of the master branch, or would it be a better idea to use gremmel's development branch for this?

- If I understood correctly, then now I just need to go through the entries in Qt linguist in order to complete the translation, right?

- How can I compile or get to run the program on my computer in my edited, latest version? Just so I can check how things look like in the running software (I am running Debian on my PC).


I pushed the edits I did so far to my github repo.


Thanks again for your help!

Cheers,
freibadschwimmer

**Comment by BourgeoisLab** [comment]
Hi freibadschwimmer

Looks good so far.
> This gave me the kb_en.ts file to work with in Qt linguist. Though it also changed a couple of other language files, hope it should be like that.

I think the other language files were not up-to-date. This should be fine.

> Is it OK if I'm working on a clone of the master branch, or would it be a better idea to use gremmel's development branch for this?

Maybe it's better to use the development branch (dev_1_4_4_x). So all texts will be up-to-date when this new version is released.
Note that the master branch can only be compiled with Qt 5.3. I started to migrate it to new Qt versions. The development branch can be compiled with Qt 5.7, although not all features are implemented yet. For translation work, you don't need to compile the program, but anyway.

> If I understood correctly, then now I just need to go through the entries in Qt linguist in order to complete the translation, right?

Actually I never did translation, but I think so.

> How can I compile or get to run the program on my computer in my edited, latest version? Just so I can check how things look like in the running software (I am running Debian on my PC).

Just copy the generated file kb_en.qm into the languages subfolder of the program. Then you can swicht the language inside the program. To show a flag icon put an en.png inside the languages folder. See attachement.
![en](https://cloud.githubusercontent.com/assets/9480307/23585969/b2093ada-018b-11e7-98f6-06f43467e2fe.png)


If you need help with translation, you can also write me directy bourgeoislab@gmail.com.

Thanks for the help. I always wanted to translate it to english but didn't find the time.

**Comment by freibadschwimmer** [comment]
Thanks for your feedback and your further comments, very much appreciated!
Alright then, I will use the dev. branch and jump into it, though as said, it might take some time.

Thanks also for offering your help for the translation work, I will come back to it :-)

If anyone else is interested (maybe @davidarduini) to contribute to the English translation please let me know, so we coordinate the work and avoid overlaps.

Cheers,
freibadschwimmer

**Comment by davidarduini** [comment]
I can assist with translation @freibadschwimmer .  I have already done some for the main page of the app.  I could create a spreadsheet or text file with the German phrases currently used, and the English equivalents as they pertain to brewing.  I have been brewing for several years, but s/w is not one of my strengths : )  Let me know how you want to proceed, and I can assist.  Thanks for all your efforts - Prost!  --Dave

**Comment by freibadschwimmer** [comment]
@davidarduini Great to hear that!
Maybe you can send me for now what you already translated/wrote down as a simple textfile (or in whatever format you have it available), just to have a look. You can send me this as an email to freibadschwimmer@kaamoscreations.com.

Thanks!
freibadschwimmer.

**Comment by davidarduini** [comment]
ok @freibadschwimmer , I am currently travelling, but I will send you what I have when I return...expect something in the next week or so.  --Dave

**Comment by freibadschwimmer** [comment]
Hi again!

I am more or less done with going through the list in QtLinguist. I have pushed the changes to my git repo.
Yet, there were a couple of words I had difficulties to translate (particularly related to the “Grünschlauchen” process). I added a page to the wiki at the “kleiner brauhelfer” and ask for your help in translating these words. The site can found here: http://www.joerum.de/kleiner-brauhelfer/doku.php?id=translation:translation

As I feel that I am in a quite advanced state of the translation process I would like to learn how I can compile the program, in order to be able to test the translations. This seems necessary, since there were quite some lines where it appears there was not just plain text but parts of the code as well. Thus I think I need to see how the translation in the compiled software looks so far in order to make the according changes.

As for now, **my questions are**:

- Should I make a pull request (at this stage) to add the changes to the dev branch?
- Can someone give me a hint how I can compile the program to run it on a Linux machine?


__________________________________________________

Just for the record: for the time being I am also willing to maintain the English translation, if corrections need to be done or when further software development requires new translations. Any help of course is highly appreciated.

Cheers,
freibadschwimmer

**Comment by BourgeoisLab** [comment]
Hi freibadschwimmer

Thank you for the help. I am quite busy at the time, so I can't help you 
with the missing translation. Maybe you can look at other brewing 
software like "brew recipe developer" http://brewrecipedeveloper.de/ .

To test the translation you don't have to compile to whole program. You 
just have to generate "kb_en.qm" with lrelease and place it together 
with "en.png" (attachment)  in the languages subfolder. See 
https://wiki.qt.io/How_to_create_a_multi_language_application#Add_translations_to_the_project

After that you should be able to change the language to English.


The easiest way to compile the program is using qtcreator.

cheers
Frederic


Am 25.06.2017 um 20:17 schrieb freibadschwimmer:

**Comment by freibadschwimmer** [comment]
> To test the translation you don't have to compile to whole program. You just have to generate "kb_en.qm" with lrelease and place it together with "en.png" (attachment) in the languages subfolder.

Thanks a lot!
Didn't know it was that easy :-)

This will certainly help to test & fine tune, but on a first glimpse it already looks OK. Will try to finish this over summer.

@davidarduini 
If you want to and have time to test the English translation in the way described above, it would be great! The relevant files can be found in the languages subfolder in my github fork.
@ Everyone: If you find any misspellings or mistakes, please let me know via email: gerald@kaamoscreations.com
Thanks!

Cheers,
freibadschwimmer

**Comment by BourgeoisLab** [comment]
Englisch translation merged.

**Comment by bitwave** [comment]
As addition, maybe it would be good to translate the homepage (and wiki/doc) aswell?

**Comment by stikkx** [comment]
i have add some translation to http://www.joerum.de/kleiner-brauhelfer/doku.php?id=translation:translation .

do you know when the english version is available?

**Comment by freibadschwimmer** [comment]
@stikkx 
Thank you so much for your inputs!
I am currently quite busy, but I will certainly add these and some minor corrections in a few weeks, when I will hopefully have some time.

Unfortunately I cannot tell you when a next version will be released, which includes the English translations. As for now you can for example use the existing translation the way @BourgeoisLab described above, (28. June).
